### PR TITLE
Remove deps: web3, lodash, bignumber.js, ajv

### PR DESCRIPTION
**Pull request by ryanio**
Also:

- Disables esModuleInterop (see reasoning: https://www.semver-ts.org/#module-interop)
- Removes fbemitter (deprecated), uses native node EventEmitter and adds browserify to create `lib/bundle.js` on build

Down to 3 deps!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,53 +1,3 @@
...
-- The `Asset` type now has `version` instead of `nftVersion` as a property
-- Similarly, the `NFTVersion` type has been renamed `TokenStandardVersion`
-- `computeFees` now takes in a single, annotated OpenSeaAsset as a parameter instead of a list of assets
-- In `isAssetTransferrable`, `didOwnerApprove` was renamed to `useProxy`
-
-Non-breaking changes with deprecation notices:
-
-- `getFungibleTokens` has been deprecated. Use `api.getPaymentTokens`
-- Methods now show a deprecation warning when used with `tokenId` or `tokenAddress` as arguments, instead of using `asset` or `assets` (of type `Asset`/`Asset[]`)
+The changelog for recent versions can be found at https://docs.opensea.io/changelog
```
Comment (JoshuaSchmidt-OpenSea) [inline CHANGELOG.md:3]: Should we also add a link to the GitHub releases?

File: package-lock.json
Comment (JoshuaSchmidt-OpenSea) [inline package-lock.json:1]: Sweet!

File: CHANGELOG.md (hunk 1 above)
Comment (ryanio) [reply CHANGELOG.md:3]: sure, will add

Comment (ryanio) [reply CHANGELOG.md]: done

File: test/sdk/orders.ts [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
-import { MAINNET_API_KEY } from "../constants";
+import { OpenSeaSDK } from "../../src/index";
+import { Network, OrderJSON } from "../../src/types";
+import { orderFromJSON } from "../../src/utils/utils";
+import { ALCHEMY_API_KEY, MAINNET_API_KEY } from "../constants";
 import ordersJSONFixture from "../fixtures/orders.json";
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const ordersJSON = ordersJSONFixture as any;
+const ordersJSON = ordersJSONFixture as unknown as OrderJSON[];
```
Comment (JoshuaSchmidt-OpenSea) [inline test/sdk/orders.ts:10]: Why does this have `as unknown as`?

File: test/integration/wrapEth.ts [hunk 3]
```diff
@@ -49,7 +47,6 @@ import { getCanonicalWrappedEther } from "../utils/tokens";
       asset: wethAsset,
     });
     const finalWeth = formatEther(finalWethBalance.toString());
-    console.log(`Final Weth: ${finalWeth}`);
     assert(
       Number(startingWeth) === Number(finalWeth),
       "Balances should match."
```
Comment (JoshuaSchmidt-OpenSea) [inline test/integration/wrapEth.ts:52]: If we are removing this console log, can we add the balances to the assert failure message?

File: test/integration/init.ts [hunk 4]
```diff
@@ -1,32 +1,30 @@
...
-);
-
 const rpcProvider = new ethers.providers.JsonRpcProvider(
   `https://eth-mainnet.g.alchemy.com/v2/${ALCHEMY_API_KEY}`
 );
 
-const wallet = new ethers.Wallet(WALLET_PRIV_KEY ?? "", rpcProvider);
+if (!WALLET_PRIV_KEY) {
+  throw new Error("WALLET_PRIV_KEY must be set for integration tests");
+}
```
Comment (JoshuaSchmidt-OpenSea) [inline test/integration/init.ts:21]: There is probably more env variables we should check here but this is still better than nothing

File: src/utils/utils.ts [hunk 5]
```diff
@@ -447,328 +337,90 @@ export const orderToJSON = (order: Order): OrderJSON => {
...
-      .call({ from: accountAddress });
-    if (params) {
-      transferFee = makeBigNumber(params[3]);
-      if (params[2] === "0") {
-        transferFeeTokenAddress = ENJIN_COIN_ADDRESS;
-      }
-    }
-  }
-  return { transferFee, transferFeeTokenAddress };
+export async function getCurrentGasPrice(provider: ethers.providers.Provider) {
```
Comment (JoshuaSchmidt-OpenSea) [inline src/utils/utils.ts:362]: I would just remove this function if it is this simple to get gas price

File: src/utils/utils.ts [hunk 6]
```diff
@@ -447,328 +337,90 @@ export const orderToJSON = (order: Order): OrderJSON => {
...
  * @param secondsToBacktrack The number of seconds to subtract on current time,
  *  to fix race conditions
- * @param shouldRoundUp Whether to round up fractional wei
  */
-export function estimateCurrentPrice(
-  order: Order,
-  secondsToBacktrack = 30,
-  shouldRoundUp = true
-) {
+export function estimateCurrentPrice(order: Order, secondsToBacktrack = 30) {
```
Comment (JoshuaSchmidt-OpenSea) [inline src/utils/utils.ts:372]: Are we sure all the numbers changed to `BigNumber.from` here are whole numbers?

File: src/utils/utils.ts [hunk 7]
```diff
@@ -338,25 +228,25 @@ export const tokenFromJSON = (token: any): OpenSeaFungibleToken => {
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export const orderFromJSON = (order: any): Order => {
```
Comment (JoshuaSchmidt-OpenSea) [inline src/utils/utils.ts:229]: Are we sure each of these are whole numbers to be able to use BIgNumber and not FixedNumber?

File: src/sdk.ts [hunk 8]
```diff
@@ -72,62 +72,48 @@ export class OpenSeaSDK {
...
-    // Ethers Config
-    this.ethersProvider = new providers.Web3Provider(
-      provider as providers.ExternalProvider
+    this.seaport_v1_5 = new Seaport(
+      this.signerOrProvider as providers.JsonRpcProvider,
+      {
+        conduitKeyToConduit: CONDUIT_KEYS_TO_CONDUIT,
+        overrides: {
+          defaultConduitKey: CROSS_CHAIN_DEFAULT_CONDUIT_KEY,
+        },
```
Comment (JoshuaSchmidt-OpenSea) [inline src/sdk.ts:107]: Are these config different than the seaport default?

File: src/sdk.ts [hunk 9]
```diff
@@ -250,10 +236,7 @@ export class OpenSeaSDK {
     amount: BigNumber,
     basisPoints: number
   ) => {
```
Comment (JoshuaSchmidt-OpenSea) [inline src/sdk.ts:238]: nit: add return type

File: src/sdk.ts [hunk 10]
```diff
@@ -455,13 +437,13 @@ export class OpenSeaSDK {
   }: {
     asset: Asset;
     accountAddress: string;
-    startAmount: BigNumberInput;
-    endAmount?: BigNumberInput;
-    quantity?: BigNumberInput;
+    startAmount: BigNumberish;
+    endAmount?: BigNumberish;
```
Comment (JoshuaSchmidt-OpenSea) [inline src/sdk.ts:441]: These amounts are gwei (whole numbers) not eth right?

File: src/sdk.ts [hunk 11]
```diff
@@ -1049,86 +1020,21 @@ export class OpenSeaSDK {
     return { basePrice, extra, paymentToken, reservePrice, endPrice };
   }
 
-  private _getSchemaName(asset: Asset | OpenSeaAsset) {
```
Comment (JoshuaSchmidt-OpenSea) [inline src/sdk.ts:1052]: Thank goodness!

File: test/sdk/orders.ts (hunk 2 above)
Comment (ryanio) [reply test/sdk/orders.ts:10]: to cast as OrderJSON[], better than any and wouldn't allow without converting to unknown first

File: test/integration/wrapEth.ts (hunk 3 above)
Comment (ryanio) [reply test/integration/wrapEth.ts:52]: yes i will swap assert for expect so it will log the difference

Comment (ryanio) [reply test/integration/wrapEth.ts:52]: actually wouldnt assert also do the same?

Comment (ryanio) [reply test/integration/wrapEth.ts:52]: oh needs assert.equal, will update

File: test/integration/init.ts (hunk 4 above)
Comment (ryanio) [reply test/integration/init.ts:21]: will add the rest

File: src/sdk.ts (hunk 8 above)
Comment (ryanio) [reply src/sdk.ts:107]: they are same, ill remove

File: src/sdk.ts (hunk 10 above)
Comment (ryanio) [reply src/sdk.ts:441]: right, wei not gwei

File: src/utils/utils.ts (hunk 7 above)
Comment (ryanio) [reply src/utils/utils.ts]: unused fn removed

File: src/utils/utils.ts (hunk 6 above)
Comment (ryanio) [reply src/utils/utils.ts:236]: removing this now-unused fn

File: test/sdk/orders.ts (hunk 2 above)
Comment (ryanio) [reply test/sdk/orders.ts]: now removed

File: test/integration/setup.ts [hunk 12]
```diff
@@ -1,32 +1,32 @@
...
   WALLET_PRIV_KEY,
-} from "../__tests__/constants";
-import { OpenSeaSDK } from "../sdk";
-import { Network } from "../types";
+} from "../utils/constants";
+
+for (const envVar of [
+  "WALLET_PRIV_KEY",
+  "SELL_ORDER_CONTRACT_ADDRESS",
+  "SELL_ORDER_TOKEN_ID",
```
Comment (JoshuaSchmidt-OpenSea) [inline test/integration/setup.ts:13]: Do we want to force someone to hold a token to run any of the integration tests? I could see someone wanting to run tests but not having a token to sell in the wallet. This would prevent them from running the other tests.

File: test/integration/postOrder.spec.ts [hunk 13]
```diff
@@ -49,7 +55,7 @@ suite("SDK: order posting", () => {
       accountAddress: walletAddress,
       amount: OFFER_AMOUNT ?? "0.004",
       quantity: 1,
-      paymentTokenAddress: WETH_ADDRESS,
+      paymentTokenAddress: WETH_ADDRESS_BY_NETWORK[Network.Main],
```
Comment (JoshuaSchmidt-OpenSea) [inline test/integration/postOrder.spec.ts:58]: Can be a separate PR but it might be nice to have Main be the default but let other networks run the test suite.

File: test/integration/README.md [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+OFFER_AMOUNT = "" # Defaults to 0.004
+LISTING_AMOUNT = "" # Defaults to 40
+```
+
+#### WETH Tests
+
+This test requires ETH and a transaction fee to be in your wallet.
+
+If you would like to run this test, you need to add `ETH_TO_WRAP = "0.001"` to your `.env` file.
+
```
Comment (JoshuaSchmidt-OpenSea) [inline test/integration/README.md:31]: Is it worth noting this has transaction costs? I don't want someone to spend ETH without realizing it.

**Review by JoshuaSchmidt-OpenSea — APPROVED** [review]

File: test/integration/setup.ts (hunk 12 above)
Comment (ryanio) [reply test/integration/setup.ts:13]: sure ill remove, itll just skip that test now
